refactor[python]: Replace pytz by stdlib zoneinfo

zundertj opened the pull request:
Dropping the third-party `pytz` package in favor of the Python stdlib `zoneinfo`. `zoneinfo` is a Python stdlib since Python 3.9. 
* For older Python versions we use `backports.zoneinfo` (https://github.com/pganssle/zoneinfo)
* `zoneinfo` relies on system timezone data. Windows does not provide that (or not in a format zoneinfo can use), so we need to install `tzdata` for that case.

See https://peps.python.org/pep-0431/ for a discussion of ZoneInfo and differences with `pytz`.

stinodego commented:
Cool, you went ahead and implemented this!

Just linking the issue for completeness: #4699

matteosantama commented on py-polars/polars/datatypes.py:
```diff
@@ -196,7 +196,8 @@ def __init__(self, time_unit: TimeUnit = "us", time_zone: str | None = None):
         time_zone
-            Timezone string as defined in pytz.
+            Timezone string as defined in zoneinfo (run
+            `import zoneinfo; zoneinfo.available_timezones()` for a full list).
```
nit: can we use double ticks here? double ticks render as code in docstrings

matteosantama reviewed (commented):
Nice!

zundertj commented:
> Cool, you went ahead and implemented this!
> 
> Just linking the issue for completeness: #4699

Oh, I added the link in the commit message body, but GitHub doesnt render that as the PR description. Thank you for spotting that.
